weblate/addons/tasks.py

```diff
@@ -159,6 +165,19 @@ def daily_callback(addon: Addon, component: Component) -> None:
     )
 
 
+@app.task(trail=False)
+def update_addon_activity_log(
+    pk: int, result: str = "", error_occurred: bool = False, pending: bool | None = None
+) -> None:
+    addon_activity_log = AddonActivityLog.objects.select_for_update().get(id=pk)
+    addon_activity_log.details["error"] = error_occurred
+    if result:
+        addon_activity_log.update_result(result)
+    if pending is not None:
+        addon_activity_log.pending = pending
+    addon_activity_log.save()
```

Comment:
Update only changed fields:

```suggestion
    addon_activity_log.save(update_fields=["details", "pending"])
```